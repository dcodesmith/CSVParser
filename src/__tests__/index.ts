import * as fs from 'fs';
import * as stream from 'stream';

import CSVParser from '..';

import { channels } from './fixtures.json';

const getCSVBuffer = (fileName: string): NodeJS.ReadableStream => {
  const bufferStream = new stream.PassThrough();
  const filePath = `${__dirname}/${fileName}.csv`;

  try {
    const buffer = fs.readFileSync(filePath);
    bufferStream.end(buffer);
  } catch (error) {
    throw error;
  }

  return bufferStream;
};

const MOCK_CHANNELS = [...channels];

test('valid', async () => {
  const CSVBuffer = getCSVBuffer('valid');

  const data = await CSVParser(CSVBuffer, MOCK_CHANNELS);

  expect(data).toMatchSnapshot();
});

test('invalid', async () => {
  const CSVBuffer = getCSVBuffer('invalid');

  try {
    await CSVParser(CSVBuffer, MOCK_CHANNELS);
  } catch (error) {
    expect(error).toBeDefined();
  }
});
