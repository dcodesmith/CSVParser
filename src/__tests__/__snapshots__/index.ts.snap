// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid 1`] = `
Array [
  Object {
    "channel": "92cb91bdc3464f14678934af",
    "date": "03/03/16",
    "description": "GEEKOL",
    "endTime": "00:45",
    "frequency": "23",
    "genre": "apple",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "News",
    "startTime": "00:00",
    "synopsis": "",
    "type": "blue",
  },
  Object {
    "channel": "92cb91bdc3464f14678934af",
    "date": "03/03/16",
    "description": "HANDSHAKE",
    "endTime": "01:45",
    "frequency": "38",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Man on Fire",
    "startTime": "00:45",
    "synopsis": "",
    "type": "brown",
  },
  Object {
    "channel": "92cb91bdc3464f14678934af",
    "date": "03/03/16",
    "description": "GEEKOL",
    "endTime": "02:15",
    "frequency": "23",
    "genre": "apple",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Escape to the Country",
    "startTime": "01:45",
    "synopsis": "",
    "type": "blue",
  },
  Object {
    "channel": "92cb91bdc3464f14678934af",
    "date": "03/03/16",
    "description": "HANDSHAKE",
    "endTime": "03:15",
    "frequency": "38",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Homes under the Hammer",
    "startTime": "02:15",
    "synopsis": "",
    "type": "brown",
  },
  Object {
    "channel": "92cb91bdc3464f14678934af",
    "date": "03/03/16",
    "description": "GEEKOL",
    "endTime": "04:00",
    "frequency": "23",
    "genre": "apple",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Football Focus",
    "startTime": "03:15",
    "synopsis": "",
    "type": "blue",
  },
  Object {
    "channel": "92cb91bdc3464f14678934af",
    "date": "03/03/16",
    "description": "HANDSHAKE",
    "endTime": "04:40",
    "frequency": "38",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Pointless",
    "startTime": "04:00",
    "synopsis": "",
    "type": "brown",
  },
  Object {
    "channel": "00cb91ccc3464f14678934da",
    "date": "03/03/16",
    "description": "SUREPLEX",
    "endTime": "00:45",
    "frequency": "39",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Boxing Hour",
    "startTime": "00:00",
    "synopsis": "",
    "type": "green",
  },
  Object {
    "channel": "00cb91ccc3464f14678934da",
    "date": "03/03/16",
    "description": "SLAMBDA",
    "endTime": "01:45",
    "frequency": "22",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "East Enders",
    "startTime": "00:45",
    "synopsis": "",
    "type": "green",
  },
  Object {
    "channel": "79cb91bdc3464f09678934ba",
    "date": "03/03/16",
    "description": "ZAJ",
    "endTime": "00:45",
    "frequency": "25",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "First Dates",
    "startTime": "00:00",
    "synopsis": "",
    "type": "brown",
  },
  Object {
    "channel": "79cb91bdc3464f09678934ba",
    "date": "03/03/16",
    "description": "ZENCO",
    "endTime": "01:45",
    "frequency": "33",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Family Guy",
    "startTime": "00:45",
    "synopsis": "",
    "type": "green",
  },
  Object {
    "channel": "92cb91bdc3464f14678934fa",
    "date": "03/03/16",
    "description": "XUMONK",
    "endTime": "00:45",
    "frequency": "35",
    "genre": "apple",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Criminal Minds",
    "startTime": "00:00",
    "synopsis": "",
    "type": "brown",
  },
  Object {
    "channel": "92cb91bdc3464f14678934fa",
    "date": "03/03/16",
    "description": "SYBIXTEX",
    "endTime": "01:45",
    "frequency": "27",
    "genre": "strawberry",
    "numberOfEpisodes": "0",
    "season": "0",
    "show": "Vinyl",
    "startTime": "00:45",
    "synopsis": "",
    "type": "green",
  },
]
`;
